disdat/db_target.py

```diff
@@ -0,0 +1,611 @@
+            rows = cursor.fetchall()
+            col_names = [x[0] for x in cursor.description]
+            df = pd.DataFrame.from_records(rows, columns=col_names)
+        return desc, df
+
+    if cursor is None:
+        with get_cursor(dsn) as cursor:
+            desc, df = _query(cursor, query)
+    else:
+        desc, df = _query(cursor, query)
+
+    if get_rows:
+        return df
+    else:
+        return desc
+
+
+def multi_tx_query(dsn, queries):
+    """
+    Issue multiple queries in a single transaction.  We depend
+    on our get_cursor() function to call cursor.commit().
+
+    Args:
+        dsn (unicode) : The database configuration dsn
+        queries (list:unicode) : Query texts
+
+    Returns:
+        list:df (pd.DataFrame) or list:desc (str)
+
+    """
+    results = []
+    with get_cursor(dsn) as cursor:
+        for q in queries:
+            results.append(single_query(dsn, q, cursor))
+
+    return results
+
+
+def table_exists_vertica(dsn, schema, table):
+    """
+    Determine whether a table exists in a Vertica DataBase
+
+    Args:
+        dsn (unicode) : The db configuration to use
+        schema (unicode) : The schema holding table
+        table (unicode) : The table name
+
+    Returns:
+        (bool) : True if table in db
+    """
+
+    query = u"""
+    SELECT DISTINCT table_name, table_type FROM all_tables
+    WHERE table_name ILIKE '{}' and
+    schema_name ilike '{}';
+    """.format(table, schema)
+
+    with get_connection(dsn) as conn:
+        result = pd.read_sql(query, conn)
+
+    if len(result) <= 0:
+        return False
+    else:
+        return True
+
+
+def schema_exists_vertica(dsn, schema):
+    """
+    Determine whether a schema exists in a Vertica DataBase
+
+    Args:
+        dsn (unicode) : The db configuration to use
+        schema (unicode) : The schema holding table
+
+    Returns:
+        (bool) : True if table in db
+    """
+
+    query = u"""
+    SELECT DISTINCT schema_name FROM schemata
+    WHERE schema_name ILIKE '{}'
+    """.format(schema)
+
+    with get_connection(dsn) as conn:
+        result = pd.read_sql(query, conn)
+
+    if len(result) <= 0:
+        return False
+    else:
+        return True
+
+
+def drop_table_vertica(dsn, table, run=True):
+    """
+
+    Args:
+        dsn:
+        table:
+        run:
+
+
+    Returns:
+
+    """
+    query = u"DROP TABLE if exists {};".format(table)
+
+    if run:
+        result = single_query(dsn, query)
+    else:
+        result = query
+
+    return result
+
+
+def drop_view_vertica(dsn, table, run=True):
+    """
+
+    Args:
+        dsn:
+        table:
+        run:
+
+
+    Returns:
+
+    """
+    query = u"DROP VIEW if exists {};".format(table)
+
+    if run:
+        result = single_query(dsn, query)
+    else:
+        result = query
+
+    return result
+
+
+def database_name_vertica(dsn):
+    """
+    SQL for getting the current database name
+
+    Returns:
+        (unicode):
```

Comment:
would be useful to fill in the Args and Returns

Reply:
Done